src/contracts/core/DelegationManager.sol

```diff
@@ -510,199 +505,178 @@ contract DelegationManager is
         delegatedTo[staker] = operator;
         emit StakerDelegated(staker, operator);
 
+        // read staker's delegatable shares and strategies to add to operator's delegatedShares
+        // and also update the staker depositScalingFactor for each strategy
         (IStrategy[] memory strategies, uint256[] memory shares) = getDelegatableShares(staker);
+        uint64[] memory totalMagnitudes = allocationManager.getTotalMagnitudes(operator, strategies);
```

Comment:
`getDelegatableShares` ends up calling `allocationManager.getTotalMagnitude` in a loop for the operator - so this is pretty inefficient.

We should probably collapse these two concepts into a single function that `returns (strategies, shares, totalMagnitudes)`. Can use an internal helper and keep the `public view` method unchanged if we need to.

Reply:
Oh mb my "go to definition" went to the wrong function since these are overloaded.